### PR TITLE
Add Ellison support

### PR DESCRIPTION
**Pull request by assertchris**
With the test server running: http://localhost:8080/?target=targets/ellison.md&wrap=1&stylesheet=stylesheets/ellison.css

Screenshot 2024-05-09 at 16 49 31

ezgif-4-c435158fc1

### CODE REVIEW

File: src/Languages/Ellison/Injections/ParserInjection.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            $sentence = preg_replace(
+                "/".preg_quote($problem['text'])."/i",
+                Escape::tokens("<span class='hl-{$problem['type']}-phrase'>").$problem['text'].Escape::tokens("</span>"),
+                $sentence,
+            );
+        }
+
+        return $sentence;
+    }
+}
```
Comment (assertchris) [inline src/Languages/Ellison/Injections/ParserInjection.php:70]: 70 lines of awesome.

File: src/Highlighter.php [hunk 2]
```diff
@@ -41,6 +42,7 @@ public function __construct(
             ->addLanguage(new CssLanguage())
             ->addLanguage(new DiffLanguage())
             ->addLanguage(new DocCommentLanguage())
+            ->addLanguage(new EllisonLanguage())
```
Comment (brendt) [inline src/Highlighter.php:45]: I'm not sure whether it should be added by default if the required package is opt-in. However, let's keep it like this for now, and we'll see whether it gives any problems :)

File: src/Languages/Ellison/Injections/ParserInjection.php (hunk 1 above)
Comment (brendt) [reply src/Languages/Ellison/Injections/ParserInjection.php:70]: This is super cool!

**Review by brendt — APPROVED** [review]
Tis is super cool! Just a great PR without any remarks, but I didn't expect anything else from you Chris :D 

Sorry for the late reply! I went down a rabbit hole in another project and wanted to solve that first before looking at my github notification again :D

**Comment by brendt** [comment]
Tagged: https://github.com/tempestphp/highlight/releases/tag/2.4.0

**Comment by brendt** [comment]
@assertchris I made a couple of small changes. The most important one is that I removed the addition of `<p>` tags. I understand why you did it, but it also meant that combining ellison support with markdown codeblock rendered `p` tags within the `pre` body, which messed up styling. 

Maybe we should make it configurable on the `EllisonLanguage`? WDYT?

**Comment by brendt** [comment]
Oh also, I added some docs about this: https://tempest.stitcher.io/highlight/04-optin

**Pull request merged**